### ISSUE TITLE
Upload error message doesn't show for format url with upload option

### ISSUE THREAD

**Issue by the-unsoul** [json-editor/json-editor#111]
in example:  `/docs/upload.html`  we had a hook:
```javascript
if (type === 'root.upload_fail') cbs.failure('Upload failed');
```
to show a custom error message when upload is failed.
But when running the example (including my attempt to use `json-editor`) none of the error is showed

**Comment by marc7000** [comment]
How did you configured your upload handler? 
The code below is an example of my implementation.

Upload handler:
```javascript
var FileUpload = function (file, cbs) {
	console.info('FileUpload: ', file, cbs);
	var form = new FormData();
	form.append('file', file);

	$.ajax({
	  type: 'POST',
	  url: '_YOUR_UPLOAD_SCRIPT_URL_',
	  data: form,
	  processData: false,
	  contentType: false,
	  xhr: function () {
		var xhr = new XMLHttpRequest();
		xhr.upload.onprogress = function (event) {
		  if (event.lengthComputable) {
			var percent = Math.round((event.loaded * 100) / event.total);
			console.info('Upload progress: ', percent);
			if (percent < 100) {
			  cbs.updateProgress(percent);
			} else if (percent === 100) {
			  cbs.updateProgress();
			}
		  }
		};
		return xhr;
	  },
	  success: function (data) {
		console.info('File uploaded: ', data, file);
		cbs.success(file.name);
	  },
	  error: function (error) {
		console.error('Upload error:', error);
		cbs.failure('File upload error...', error);
	  }
	});
};
```

JSONEditor configuration:
```javascript
JSONEditor.defaults.options.upload = function (type, file, cbs) {
  if (file) {
	new FileUpload(file, cbs);
  }
};
```

**Comment by the-unsoul** [comment]
@marc7000 

at original my upload handler was exactly like the example in `/docs/upload.html`

```javascript
JSONEditor.defaults.options.upload = function (type, file, cbs) {
    if (type === 'root.upload_fail') {
        cbs.failure('Upload failed');
    } else {
        if (file) {
            if (file.size/8 <= 512) {
                var FR = new FileReader();
                
                FR.addEventListener("load", function(e) {
                    cbs.success(e.target.result);
                    // document.getElementById("b64").innerHTML = e.target.result;
                }); 
                
                FR.readAsDataURL( file );
            } else {
                console.log(editor.validation_results);
                editor.root.showValidationErrors(editor.validation_results);
                cbs.failure('Upload failed: image must be smaller than 500KB');
            }
        }
    }
};
```

In short the upload process working as I wanted it to be.
BUT ONLY EXCEPT the part that I error message doesn't show when I wanted it to show `cbs.failure('Upload failed');`

**Comment by marc7000** [comment]
How looks your JSON Schema? 
Do you have a property "upload_fail"? And if so, why?

**Comment by the-unsoul** [comment]
@marc7000 ,
As said I was using the example in `/docs/upload.html`

So yes, my scheme is exactly like in the example, and I have that property (plus with some additional logic to check if the file is bigger than 500kb)

So here is my schema (same exact scheme in `/docs/upload.html`)
```javascript
var editor = new JSONEditor(document.getElementById('editor_holder'), {
    schema: {
        type: "object",
        title: "Image",
        properties: {
            upload_default: {
                type: "string",
                format: "url",
                options: {
                    upload: true
                },
                "links": [{
                    "href": "{{self}}"
                }]
            },
            upload_custom_link: {
                type: "string",
                format: "url",
                options: {
                    upload: true
                },
                "links": [{
                    "href": "{{self}}",
                    "rel": "view"
                }]
            },
            upload_readonly: {
                readonly: true,
                type: "string",
                format: "url",
                options: {
                    upload: true
                },
                "links": [{
                    "href": "{{self}}"
                }]
            },
            upload_fail: {
                type: "string",
                format: "url",
                options: {
                    upload: true
                },
                "links": [{
                    "href": "{{self}}"
                }]
            },
            name: {
                type: "string"
            }
        }
    }
});
```

As to this question
> And if so, why?
It is to see if error message display or not, so I can put some custom error message.

P.S. as noted, even the example `/docs/upload.html` doesn't work, so please don't concern how my code is, as you can easily reproduce it by just running the example :)

**Comment by marc7000** [comment]
As you can see, the JSON-Editor upload example works as desired.
![json-editor-upload](https://user-images.githubusercontent.com/1118837/43908356-16ebce42-9bf8-11e8-9672-6b89e7fed39c.gif)

**Comment by the-unsoul** [comment]
UPDATE:

The issue seem to be still there, and it is only reproduce when a theme is selected (e.g bootstrap, foundation...)

**Comment by pmk65** [comment]
It's because the `controlgroup` is not defined. The controlgroup defines which element to place the error message. Without it no error message.
This have been fixed in #607 #607
